### PR TITLE
Fixup negative from days

### PR DESCRIPTION
**Pull request by satanson**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #
https://github.com/StarRocks/starrocks/issues/4720

## Problem Summary(Required) ：

In MySQL 8.0,  when the argument ranges from  0 ~ 3652424(include), from_days gives a valid date('0000-01-01') ~ ('9999-12-31'); otherwise(the argument below 0 or above 3652424), from_days gives date('0000-00-00');
i.e:
select from_days(-1) should return "0000-00-00"
select from_days(3652425) should return "0000-00-00"

### CODE REVIEW

File: be/src/runtime/time_types.h [hunk 1]
```diff
@@ -375,6 +377,13 @@ const constexpr uint32_t CACHE_JULIAN_DAYS = 200 * 366;
 extern JulianToDateEntry g_julian_to_date_cache[];
 
 inline void date::to_date(JulianDate julian, int* year, int* month, int* day) {
+    if (julian == ZERO_EPOCH_JULIAN) {
```
Comment (kangkaisen) [inline be/src/runtime/time_types.h:380]: UNLIKELY

File: be/src/types/timestamp_value.h [hunk 2]
```diff
@@ -109,6 +109,8 @@ class TimestampValue {
 
     bool is_valid() const;
 
+    bool is_valid_non_strict() const;
```
Comment (kangkaisen) [inline be/src/types/timestamp_value.h:112]: comment the difference between from `is_valid` and `is_valid_non_strict`.

**Review by murphyatwork — DISMISSED** [review]

File: be/src/types/timestamp_value.h (hunk 2 above)
Comment (satanson) [reply be/src/types/timestamp_value.h:112]: for DateValue "0000-00-00", is_valid return false, is_valid_non_strict return true;

**Review by kangkaisen — APPROVED** [review]

**Review by silverbullet233 — APPROVED** [review]

**Review by murphyatwork — APPROVED** [review]

**Comment by satanson** [comment]
@MergifyIO backport branch-2.3

**Pull request merged**